### DISCUSSION TITLE
Possible flood atack to fallback: true

### DISCUSSION THREAD

**Discussion by hacknlove** [vercel/next.js#12916] [Help] [answered]
What if someone gets to know that some of your urls are dinamic SSG, and he makes your server to generet and cache billions of pages?

For instance the url `/product/[id]` and He just uses random ids.

Are we protected against this kind of attack?

**Comment by ijjk** [comment]
Hi, this isn’t a concern when deploying on [Vercel](https://vercel.com), although when deploying with custom infrastructure you can disable writing to the filesystem by adding `experimental: { sprFlushToDisk: false }` to your `next.config.js` which will then use an in-memory LRU cache only. You could also implement custom filesystem pruning for seldom used files if desired. 

The most streamlined approach for leveraging this functionality would be deploying on [Vercel](https://vercel.com)

**Reply by hacknlove** [reply]
I'm deploying on vercel, and I am paying even though the free tier is enough for me, because I want to give them back something. *now* is awesome, *nextjs* is awesome, and I hope they keep building awesomic stuff.   

But anyway I'm a bit concerned.

What if there is a political site on vercel, and someone makes a big scale attack on this.

Could other sites be affected?

Is this site going offline? Is vercel offlining this site to prevent issues on the rest of the sites?

BTW, I am doing a activism site, I expect having 400k users in a online protests, and I expect also to suffer attacs by contrary party.

**Reply by ijjk** [reply]
Awesome, thanks for the support! We do take many security measures on Vercel to prevent attacks to sites like this and others. If you have any specific concerns beyond this you can reach out to support@vercel.com for more information

**Comment by Kodpact** [comment, marked as answer]
Pass notFound inside getStaticProps, if its true then page wont be generated.
Like:
`return {
			props: { data: null },
			revalidate: 20,
			notFound: true,
		};`

Note that revalidate wont work if notFound returns true!
Currently it is locked on 1 second.
